src/test/java/bio/terra/service/dataset/DatasetConnectedTest.java

```diff
@@ -507,8 +507,128 @@ public void testRepeatedSoftDelete() throws Exception {
+        connectedOperations.softDeleteSuccess(summaryModel.getId(), softDeleteRequest); // TODO This is what is breaking
+        connectedOperations.softDeleteSuccess(summaryModel.getId(), softDeleteRequest); // TODO This is what is breaking
+        connectedOperations.softDeleteSuccess(summaryModel.getId(), softDeleteRequest); // TODO This is what is breaking
+        connectedOperations.softDeleteSuccess(summaryModel.getId(), softDeleteRequest); // TODO This is what is breaking
+        connectedOperations.softDeleteSuccess(summaryModel.getId(), softDeleteRequest); // TODO This is what is breaking
+        connectedOperations.softDeleteSuccess(summaryModel.getId(), softDeleteRequest); // TODO This is what is breaking
+        connectedOperations.softDeleteSuccess(summaryModel.getId(), softDeleteRequest); // TODO This is what is breaking
+        connectedOperations.softDeleteSuccess(summaryModel.getId(), softDeleteRequest); // TODO This is what is breaking
+        connectedOperations.softDeleteSuccess(summaryModel.getId(), softDeleteRequest); // TODO This is what is breaking
+        connectedOperations.softDeleteSuccess(summaryModel.getId(), softDeleteRequest); // TODO This is what is breaking
+        connectedOperations.softDeleteSuccess(summaryModel.getId(), softDeleteRequest); // TODO This is what is breaking
+        connectedOperations.softDeleteSuccess(summaryModel.getId(), softDeleteRequest); // TODO This is what is breaking
+        connectedOperations.softDeleteSuccess(summaryModel.getId(), softDeleteRequest); // TODO This is what is breaking
+        connectedOperations.softDeleteSuccess(summaryModel.getId(), softDeleteRequest); // TODO This is what is breaking
+        connectedOperations.softDeleteSuccess(summaryModel.getId(), softDeleteRequest); // TODO This is what is breaking
+        connectedOperations.softDeleteSuccess(summaryModel.getId(), softDeleteRequest); // TODO This is what is breaking
+        connectedOperations.softDeleteSuccess(summaryModel.getId(), softDeleteRequest); // TODO This is what is breaking
+        connectedOperations.softDeleteSuccess(summaryModel.getId(), softDeleteRequest); // TODO This is what is breaking
+        connectedOperations.softDeleteSuccess(summaryModel.getId(), softDeleteRequest); // TODO This is what is breaking
+        connectedOperations.softDeleteSuccess(summaryModel.getId(), softDeleteRequest); // TODO This is what is breaking
+        connectedOperations.softDeleteSuccess(summaryModel.getId(), softDeleteRequest); // TODO This is what is breaking
+        connectedOperations.softDeleteSuccess(summaryModel.getId(), softDeleteRequest); // TODO This is what is breaking
+        connectedOperations.softDeleteSuccess(summaryModel.getId(), softDeleteRequest); // TODO This is what is breaking
+        connectedOperations.softDeleteSuccess(summaryModel.getId(), softDeleteRequest); // TODO This is what is breaking
+
```

Comment:
I had to run this a few times to get it to break the first time and then it would break earlier and earlier in the test until it was breaking on like the third call